### PR TITLE
Bug 1223993 - migrate ringtone app to NotificationHelper

### PR DESCRIPTION
**Pull request by hkasemir**
https://bugzilla.mozilla.org/show_bug.cgi?id=1223993

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ae9429d9764483326b73f0b15a2b60d17167ea97

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7c4a5a2727a1e2b90389344ee1ea5e618bd0daed

File: apps/ringtones/js/share.js [hunk 1]
```diff
@@ -228,13 +227,15 @@ function handleShare(data, callback) {
         // Show a notification indicating success, and then close it immediately
         // so it doesn't stink up the notifications tray! XXX: This UX isn't
         // great; we should turn this into a transient notification when we can.
-        new Notification(
+        NotificationHelper.send(
           songtitle,
-          {body: navigator.mozL10n.get(
-            details.setAsDefault ? 'set-default-tone' : 'created-tone'
-          )}
-        ).close();
```
Comment (hkasemir) [inline apps/ringtones/js/share.js]: does this mean that closeOnClick should be left to default `true` value?

Comment (zbraniecki) [inline apps/ringtones/js/share.js]: no, it should be `closeOnClick: false` here

File: apps/ringtones/js/share.js [hunk 2]
```diff
@@ -228,13 +227,15 @@ function handleShare(data, callback) {
...
-        callback('save', details);
+          { bodyL10n:
+            details.setAsDefault ? 'set-default-tone' : 'created-tone',
+            closeOnClick: false
+          }
+        ).then(() => {
```
Comment (zbraniecki) [inline apps/ringtones/js/share.js]: and here you need:

```javascript
.then(notification => {
  notification.close();
  callback('save', details);
});
```

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8523880486d2a40c1137de28187abe6faeca8fe9

File: apps/ringtones/js/share.js [hunk 3]
```diff
@@ -228,13 +227,16 @@ function handleShare(data, callback) {
...
-        new Notification(
+        NotificationHelper.send(
           songtitle,
-          {body: navigator.mozL10n.get(
-            details.setAsDefault ? 'set-default-tone' : 'created-tone'
-          )}
-        ).close();
-        callback('save', details);
+          { bodyL10n:
+            details.setAsDefault ? 'set-default-tone' : 'created-tone',
```
Comment (jimporter) [inline apps/ringtones/js/share.js]: Nit: indent this by 2 more spaces, please. Or move some of it onto the previous line, e.g.:

```
          { bodyL10n: details.setAsDefault ? 'set-default-tone' :
                      'created-tone',
```

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d9ea2d23207d90c2ba8a5358a3ba31987cb31ad7

**Pull request merged**